index.html

```diff
@@ -1341,43 +1363,89 @@ <h4>Rename Blank Nodes</h4>
+                </li>
+              </ol>
+            </li>
+            <li>Otherwise, if <var>element</var> is a <a>map</a>:
+              <ol>
+                <li>Initialize an empty <a>map</a> <var>result</var>.</li>
+                <li>For each <var>key</var>-<var>value</var> pair in <var>element</var>:
+                  <ol>
+                    <li>If <var>key</var> is `@id` and <var>value</var> if a <a>blank node identifier</a>,
+                      add an <a>entry</a> to <var>result</var> for `@id` with the value from the result of the
+                      <a data-cite="JSON-LD11-API#generate-blank-node-identifier">Generate Blank Node Identifier algorithm</a>
+                      passing <var>value</var> for <var>identifier</var>.</li>
+                    <li>Otherwise, add an <a>entry</a> to <var>result</var> for `@id`
+                      with the value from the result of calling this algorithm recursively
+                      using <var>value</var> for <var>element</var>.</li>
+                  </ol>
+                </li>
+              </ol>
+            </li>
+            <li>Otherwise, set <var>result</var> to <var>element</var></li>
+          </ol>
+
+          <p>Return <var>result</var>.</p>
+      </section>
+
+      <section id="create-annotations" class="algorithm">
+        <h4>Create Annotations</h4>
+
+        <p>This algorithm is used to find entries in a <em>node map</em> where
+          the key is a canonicalized <a>embedded node</a> and that
+          <a>embedded node</a> exists in non-embedded form
+          in the <em>node map</em>
+          and move them to be <a>annotation objects</a> under that node.
+          This effectively reverses the process of unfolding
+          <a>annotation objects</a> performed in
+          <a href="#node-map-generation" class="sectionRef"></a>.
+          It is also used in <a href="#serialize-rdf-as-json-ld" class="sectionRef"></a>.</p>
 
+        <p>The algorithm takes a single input variable <var>node map</var>
+          and updates it in place.</p>
+
+        <ol>
+          <li>For each <a>entry</a> <var>key</var>/<var>annotation</var> in <var>node map</var>,
+            where the first character of <var>key</var>
+            is an `OPEN BRACE` (`{`), reverse ordered by length:
             <ol>
-              <li>If <var>element</var> is an <a>array</a>:
-                <ol>
-                  <li>Initialize an empty <a>array</a> <var>result</var>.</li>
-                  <li>For each <var>item</var> in <var>element</var>
-                    add the result of invoking this algorithm recursively
-                    passing <var>item</var> for <var>element</var>.
-                  </li>
-                </ol>
-              </li>
-              <li>Otherwise, if <var>element</var> is a <a>map</a>:
+              <li>Deserialize <var>key</var> into a new
+                <a>map</a> <var>embedded</var>.</li>
```

Comment:
Why not simply getting the `@id` entry of <var>annotation</var>? Shouldn't this be equivalent?

Reply:
Yes, probably. I'll verify in my own code base.

Reply:
Addressed in 311fce8e.